protoc-gen-grpc-gateway/gengateway/template_test.go

```diff
@@ -402,3 +403,37 @@ func TestApplyTemplateRequestWithClientStreaming(t *testing.T) {
 		}
 	}
 }
+
+func Test_queryParamFilter_String(t *testing.T) {
+	tests := []struct {
+		name string
+		seqs [][]string
+		want string
+	}{
+		{
+			name: "Empty queryParamFilter",
+			seqs: [][]string{},
+			want: "&utilities.DoubleArray{Encoding: map[string]int{}, Base: []int(nil), Check: []int(nil)}",
+		},
+		{
+			name: "Simple queryParamFilter",
+			seqs: [][]string{{"foo"}},
+			want: `&utilities.DoubleArray{Encoding: map[string]int{"foo": 0}, Base: []int{1, 1, 0}, Check: []int{0, 1, 2}}`,
+		},
+		{
+			name: "Full queryParamFilter",
+			seqs: [][]string{{"foo", "bar"}, {"foo", "baz"}, {"qux"}},
+			want: `&utilities.DoubleArray{Encoding: map[string]int{"foo": 0, "bar": 1, "baz": 2, "qux": 3}, Base: []int{1, 1, 1, 2, 3, 0, 0, 0}, Check: []int{0, 1, 2, 2, 1, 3, 4, 5}}`,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			f := queryParamFilter{
+				DoubleArray: utilities.NewDoubleArray(tt.seqs),
```

Comment:
I could remove the call to NewDoubleArray and set the fields directly so that only code in the template.go file is tested.